server/services/docs.ts

```diff
@@ -0,0 +1,99 @@
+import { readdirSync, readFileSync, statSync } from 'fs'
+import matter from 'gray-matter'
+import { nanoid } from 'nanoid'
+import { DocFileOrHeading, DocsMenuContent } from '../../types/interfaces'
+
+type DocsFrontMatter = {
+  priority?: number
+}
+
+const sortByPriority = (a: DocFileOrHeading, b: DocFileOrHeading): number => {
```

Comment:
I think this can be more trivially implemented as `return (a.priority || Infinity) - (b.priority || Infinity)`, unless I'm missing something.

Reply:
I would usually suggest making `priority` a guaranteed value on Doc with a value of `Infinity`, but if I remember correctly it doesn't have a JSON representation beyond `null` which is pretty annoying.

Reply:
I've changed priority to be a required property of `DocHeading` and `DocFile` that defaults to `Number.MAX_SAFE_INTEGER` if not in the front matter, so this has been simplified.